src/push-alert.ts

```diff
@@ -0,0 +1,78 @@
+import * as github from '@actions/github'
+import { GitHub } from '@actions/github/lib/utils';
+import * as core from '@actions/core';
+var request = require('request');
+
+export async function runPushAlert(args) {
+    try {
+      if (!args.slackEndpoint){
+        throw new Error('Slack notification endpoint undefined');
+      }
+      // this is requried for List branches or pull requests for a commit
+      // detail: https://developer.github.com/v3/previews/#list-branches-or-pull-requests-for-a-commit
+      const client = github.getOctokit(args.repoToken);
+      //const commits = github.context.payload.commits
+      if (!process.env.GITHUB_SHA){
+        return;
+      }
+      const commits = [process.env.GITHUB_SHA];
```

Comment:
am I reading correct `commits` is always array with length 1?

Reply:
oh yeah I should have simply this code by removing the for loop below
but to answer your question, yeah we only look at the head of the push since the approval status is only associated with that sha

Reply:
yes, q was about below loop I felt odd since there seems only one item and we try to loop over.